crates/pallets/starknet/src/simulations.rs

```diff
@@ -45,22 +45,26 @@ impl<T: Config> Pallet<T> {
             .into_iter()
             .map(|tx| match Self::execute_account_transaction(&tx, &mut state, &block_context, simulation_flags) {
                 Ok(execution_info) if !execution_info.is_reverted() => Ok(execution_info),
-                Err(_) | Ok(_) => Err(Error::<T>::TransactionExecutionFailed),
+                Ok(execution_info) => {
+                    Err(Error::TransactionExecutionFailed(execution_info.revert_error.unwrap().to_string()))
+                }
+                Err(e) => Err(Error::from(e)),
             })
             .map(|exec_info_res| {
                 exec_info_res.map(|exec_info| {
                     exec_info
                         .actual_resources
                         .0
                         .get("l1_gas_usage")
-                        .ok_or_else(|| DispatchError::from(Error::<T>::MissingL1GasUsage))
+                        .ok_or_else(|| DispatchError::from(ConfigError::<T>::MissingL1GasUsage))
```

Comment:
should we make this an simulation error as well so its communicated to the client?